src/Microsoft.OData.Edm/Csdl/Serialization/EdmModelCsdlSchemaXmlWriter.cs

```diff
@@ -436,6 +436,23 @@ internal override void WriteCollectionTypeElementHeader(IEdmCollectionType colle
             }
         }
 
+        internal static bool IsUsingDefaultValue(IEdmVocabularyAnnotation annotation)
+        {
+            EdmVocabularyAnnotation edmAnnotation = annotation as EdmVocabularyAnnotation;
+            if (edmAnnotation != null)
+            {
+                return edmAnnotation.UseDefault;
+            }
+
+            CsdlSemanticsVocabularyAnnotation csdlAnnotation = annotation as CsdlSemanticsVocabularyAnnotation;
+            if (csdlAnnotation != null)
+            {
+                return csdlAnnotation.UseDefault;
+            }
+
+            return false;
+        }

```

Comment:
I don't love this logic, because it special cases the EdmVocabularyAnnotation and CsdlSemanticsVocabularyAnnotation, meaning that a custom implementation of IEdmVocabularyAnnotation can't implement UseDefault; however, I am okay putting it here as long as we fix it by adding usesDefault to IEdmVocabularyAnnotation in 8.0.  @xuzhg -- can you open an issue, tagged 8.0/breaking change opportunity, to address this there?

Reply:
Done at: https://github.com/OData/odata.net/issues/2228